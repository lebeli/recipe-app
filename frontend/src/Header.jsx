import React, { Component } from 'react';
import './Header.scss';
import Box from '@material-ui/core/Box';

import img from './images/foodbaby.png';

class Header extends Component {
  render() {
    return (
      <div className="Header">
        <Box display="flex" justifyContent="center" m={1} p={1}>
          <img id="logo" src={img} alt="foodbaby logo" />
        </Box>
        <Box display="flex" justifyContent="center" m={1} p={1}>
          <p id="slogan">Was isst du heute?</p>
        </Box>
      </div>
    );
  }
}

export default Header;
