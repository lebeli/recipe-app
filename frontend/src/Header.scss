@import './styles/styles.scss';

.Header {
    background-color: $tertiaryColor;

    .MuiBox-root {
        margin: 0;
    }

    #logo {
        width: 130px;
    }

    #slogan {
        font-family: $primaryFont;
        font-size: 16pt;
        color: white;
        margin: 0 0 20px 0;
    }
}