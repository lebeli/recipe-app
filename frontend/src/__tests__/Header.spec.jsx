import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import Header from "../Header";

describe("<Header />", () => {
  test("should render elements", () => {
    const wrapper = shallow(<Header />);
    expect(wrapper.find(".Header"));
  });

  test("renders correctly", () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
