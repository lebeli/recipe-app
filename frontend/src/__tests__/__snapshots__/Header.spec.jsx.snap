// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly 1`] = `
<div
  className="Header"
>
  <div
    className="MuiBox-root MuiBox-root-2"
  >
    <img
      alt="foodbaby logo"
      id="logo"
      src="test-file-stub"
    />
  </div>
  <div
    className="MuiBox-root MuiBox-root-3"
  >
    <p
      id="slogan"
    >
      Was isst du heute?
    </p>
  </div>
</div>
`;
